core/src/main/java/org/jdbi/v3/core/mapper/PrimitiveMapperFactory.java

```diff
@@ -59,7 +60,14 @@
     }
 
     private static <T> ColumnMapper<T> primitiveMapper(ColumnGetter<T> getter) {
-        return (r, i, ctx) -> getter.get(r, i);
+        return (r, i, ctx) -> {
+            T value = getter.get(r, i);
+            if (r.wasNull() && !ctx.getConfig(ColumnMappers.class).getNullPrimitivesToDefaults()) {
+                throw new UnableToProduceResultException("a Java primitive value was attempted to be mapped from a database null value");
```

Comment:
If there are multiple primitives columns, users won't be able to tell which column failed to map from this error message. We should provide more information about which column failed to be mapped. Maybe include the column number (and column label from the result set metadata)?

We may also want to include a suggestion in the error message to change the mapped type from a primitive to a wrapper.